b2sdk/account_info/abstract.py

```diff
@@ -99,6 +100,23 @@ def clear_bucket_upload_data(self, bucket_id):
         :param str bucket_id: a bucket ID
         """
 
+    def check_current_credentials(self, key_id, key, realm):
```

Comment:
The method name kind of suggests a `void` type that would raise an exception if something is wrong, we do follow such convention in the project already with checking key permissions, I think.

I think it should be renamed, at least, but usually `bool` methods names start with `is_`, while here it doesn't stick because we are checking two things...

A similar method should in the future be used to check if we should clean the cache when reauthorizing. If the account_id is the same as it was and realm is the same as it was, then it's just a different key for the same account, not need to flush the cache, right? Will we add another method to check current account? And then every time we will want to check something in the account info we will add a method to the account info interface?

Perhaps
```python
try:
    should_authorize = account_info.get_application_key_id() == key_id and account_info.get_realm() == realm
except MissingAccountInfo:
    should_authorize = False
```
is something that could just live in the client that tries to compare those things?

Reply:
Naming:

I agree with the `is_` convention. I have done methods named `are_` in the past, but these get confusing sometimes. With your feedback, I''d go with `is_auth_data_correct`.

Moving the logic into the client:
That's something I was torn about. It would be way simplier, especially taking into account that the CLI and the SDK are two repos. But it just seems like such a method SHOULD live in the SDK, it's the SDKs domain. Taking into account the project's high code quality, I didn't want to go for the low hanging fruit. But If you think it's ok to put this into the CLI then I'm with you, and I'll close this PR.

Reply:
In `session.py` we have now:
```python
        # Clean up any previous account info if it was for a different account.
        try:
            old_account_id = self.account_info.get_account_id()
            old_realm = self.account_info.get_realm()
            if application_key_id != old_account_id or realm != old_realm:
                self.cache.clear()
        except MissingAccountData:
            self.cache.clear()
```

When following the rule that the best code should be self-descriptive and shouldn't require too many comments, wouldn't be better for a reader to write it like:

```python
        if not self.account_info.is_the_same_account(application_key_id, realm):
            self.cache.clear()
```
?

Reply:
kind of, but in the case of b2cli you want to start using the new key even if it is from the same account, so `is_same_account(application_key_id, realm)` is good for `session.py`, but cli needs ``is_same_key(application_key_id, realm)`.

@mpnowacki-reef please implement both methods and also do the changes in `session.py` (and add tests)

Reply:
Okay, I'll do that. But analysing the code, I came to a conclusion, that the mentioned logic in `session.py` is flawed, because `application_key_id` is compared to `self.account_info.get_account_id()` which are different values, although related, and one of these strings is contained by the other. In fact, this method accepts a `key_id` and `key`, so this is not a case for `is_same_account` but for `is_same_key`.

Reply:
I think I'll add a test that checks that `cache.clear` is not called when running `authorize_account` two consecutive times

Reply:
Having the situation where `is_same_key() == True` forced `is_same_account() == True` and I belive in `B2Session.authorize_account` we may want to check for key, not only the account. So we don't need `is_same_account` for now, unless there is some other part of the code we could use it.

@mpnowacki-reef good catch

Reply:
I think so too, I wasn't going to implement `is_same_account` (though from comment anyone would assume that I would)